draft-ietf-idr-linklocal-capability.xml

```diff
@@ -180,77 +162,224 @@ next hops or IPv6 link-local only next hop.</t>
       <name>Error handling</name>
-      <t>A BGP speaker receiving an MP_REACH_NLRI with the length of the
-Next Hop Field set to 32, where the update contains anything other than
-an IPv6 link-local address and a global address, SHOULD consider
-this a malformed UPDATE message, and proceed as described in the following
-paragraphs. In order to support backward compatibility with existing
-implementations, an implementation MAY ignore a second IPv6 link-local address
-or 0::0/0 included with an IPv6 link-local address when the length of
-the Next Hop Field is set to 32; in this case, the implementation SHOULD
-report the existence of this additional information so the operator can
-correct the sending BGP implementation.</t>
-      <t>And if a BGP speaker receiving an MP_REACH_NLRI with the length
-of the Next Hop Field set to 16, where the update contains an IPv6 link-local
-address, SHOULD consider this a malformed UPDATE message, and handle
-the malformed UPDATE message using the approach of "treat-as-withdraw",
-as described in section 7.3 of <xref target="RFC7606"/>.</t>
-      <t>If the Next Hop field is malformed, the implementation MUST handle the
-malformed UPDATE message using the approach of "treat-as-withdraw", as
-described in section 7.3 of <xref target="RFC7606"/>.</t>
-      <t>If the Next Hop field is properly formed, but the IPv6 link-local next hop is not
-reachable (as determined by an examination of the IPv6 neighbor table), the
-implementation MAY handle the malformed UPDATE message using the approach
-of "treat-as-withdraw", as described in section 7.3 of <xref target="RFC7606"/>
-(see the note above on checking the local neighbor table for the correctness of
-the next hop).</t>
+      <t>
+        These procedures apply only when the Link-Local Next Hop Capability has
+        been negotiated for a BGP session.
+      </t>
+      <t>
+        A BGP speaker receiving an MP_REACH_NLRI with the length of the Next Hop
+        Field set to 32, where the update contains anything other than a Global IPv6
+        followed by a Link-Local IPv6 address, SHOULD do the following.
+      </t>
+      <t>
+        When both IPv6 next hops contain Link-Local IPv6 addresses, the second
+        Link-Local IPv6 address should be used for forwarding.  If both address
+        are not identical, the implementation should bring this to the
+        operator's attention for debugging.
+      </t>
+      <t>
+        When the first IPv6 address is 0::0/0 and the second IPv6 address is an
+        IPv6 Link-Local address, the second address is used for forwarding.
+      </t>
+      <t>
+        If the Next Hop field is malformed, the implementation MUST handle the
+        malformed UPDATE message using the approach of "treat-as-withdraw", as
+        described in section 7.3 of <xref target="RFC7606"/>.
+      </t>
+      <t>
+        <!-- XXX JMH - This section is problematic for third party LLNH.  For
```

Comment:
I believe this should be handled by the next-hop tracking (reachability), and we shouldn't refer to it as "malformed"?

Reply:
I'll pinged Russ, and Jeff on Slack regarding this PR (to review).